File: backend/src/main/java/com/likelion/yonsei/baton/domain/classification/entity/Classification.java
package com.likelion.yonsei.baton.domain.classification.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.JdbcTypeCode;
import org.hibernate.type.SqlTypes;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "classifications")
public class Classification {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;

	@Column(name = "baton_id", nullable = false)
	private Long batonId;

	@Column(name = "reply_message_id", nullable = false)
	private Long replyMessageId;

	@Column(name = "selected_branch_id")
	private Long selectedBranchId;

	@Column(precision = 5, scale = 4)
	private BigDecimal confidence;

	@Column(name = "is_ambiguous", nullable = false)
	private boolean ambiguous;

	@Column(name = "contains_new_question", nullable = false)
	private boolean containsNewQuestion;

	@JdbcTypeCode(SqlTypes.JSON)
	@Column(name = "extracted_data_json", columnDefinition = "jsonb")
	private String extractedDataJson;

	@Column(name = "reasoning_summary", columnDefinition = "TEXT")
	private String reasoningSummary;

	@Enumerated(EnumType.STRING)
	@Column(name = "result_status", nullable = false, length = 30)
	private ClassificationResultStatus resultStatus;

	@Column(name = "model_name", length = 100)
	private String modelName;

	@CreationTimestamp
	@Column(name = "created_at", nullable = false, updatable = false)
	private LocalDateTime createdAt;

	protected Classification() {
	}

	public Classification(
			Long batonId,
			Long replyMessageId,
			Long selectedBranchId,
			BigDecimal confidence,
			boolean ambiguous,
			boolean containsNewQuestion,
			String extractedDataJson,
			String reasoningSummary,
			ClassificationResultStatus resultStatus,
			String modelName
	) {
		this.batonId = batonId;
		this.replyMessageId = replyMessageId;
		this.selectedBranchId = selectedBranchId;
		this.confidence = confidence;
		this.ambiguous = ambiguous;
		this.containsNewQuestion = containsNewQuestion;
		this.extractedDataJson = extractedDataJson;
		this.reasoningSummary = reasoningSummary;
		this.resultStatus = resultStatus;
		this.modelName = modelName;
	}

	public void markGuardrailRejected() {
		this.resultStatus = ClassificationResultStatus.GUARDRAIL_REJECTED;
	}

	public Long getId() {
		return id;
	}

	public Long getBatonId() {
		return batonId;
	}

	public Long getReplyMessageId() {
		return replyMessageId;
	}

	public Long getSelectedBranchId() {
		return selectedBranchId;
	}

	public BigDecimal getConfidence() {
		return confidence;
	}

	public boolean isAmbiguous() {
		return ambiguous;
	}

	public boolean isContainsNewQuestion() {
		return containsNewQuestion;
	}

	public String getExtractedDataJson() {
		return extractedDataJson;
	}

	public String getReasoningSummary() {
		return reasoningSummary;
	}

	public ClassificationResultStatus getResultStatus() {
		return resultStatus;
	}

	public String getModelName() {
		return modelName;
	}

	public LocalDateTime getCreatedAt() {
		return createdAt;
	}
}

File: frontend/my-app/src/types/classification.ts
import type { ClassificationResultStatus } from './enums'

/**
 * classifications 중 프론트에서 쓰는 부분.
 * model_name은 운영/디버그용이라 제외. extracted_data_json은 용도가 불명확해 제외 —
 * 필요해지면 스키마 논의와 함께 다시 추가.
 *
 * isAmbiguous / containsNewQuestion 두 값이 PendingResponse 화면의 두 상태를 가른다:
 *   - isAmbiguous만 true       → 후보 분기 중 선택
 *   - containsNewQuestion true (또는 위 둘 다 false — NO_MATCH/GUARDRAIL_REJECTED 등)
 *     → 분기 선택 대신 상대의 새 질문 강조 + 직접 작성 유도
 *
 * replyMessageId는 실제 API 응답에 없다 — 답장 메시지는 baton.replyMessageId로 조회한다.
 *
 * candidateBranchIds는 실제 API 응답에도 없는 필드다(백엔드 API 명세서 확인 완료 —
 * "후보 분기 2개"를 저장/응답할 컬럼 자체가 없음). is_ambiguous일 때 어떤 분기가
 * 후보였는지 구분할 방법이 없어서, 화면에서는 해당 바통의 branches 전체를 후보로
 * 보여주는 걸로 대체한다 (docs/api-integration.md 참고). 항상 null.
 */
export interface Classification {
  id: string
  batonId: string
  selectedBranchId: string | null
  candidateBranchIds: string[] | null
  confidence: number | null
  isAmbiguous: boolean
  containsNewQuestion: boolean
  reasoningSummary: string | null
  resultStatus: ClassificationResultStatus
}
